examples/linear_model/plot_ridge_coeffs.py

```diff
@@ -1,89 +1,272 @@
+:class:`~sklearn.linear_model.Ridge` regression affects a model's performance by
+adding a penalty term to the Loss that increases with the coefficients (β).
+
+The regularized loss function is given by: :math:`\\mathcal{L}(X, y, \\beta) = \\|
+y - X \\beta \\|^{2}_{2} + \\alpha \\| \\beta \\|^{2}_{2}`
+
+where :math:`X` is the input data, :math:`y` is the target variable,
+:math:`\\beta` is the vector of coefficients associated with the features, and
+:math:`\\alpha` is the regularization strength.
+
+The regularized loss function aims to balance the trade-off between accurately
+predicting the training set and to prevent overfitting.
+
+In this regularized loss, the left-hand side (:math:`\\|y - X\\beta\\|^{2}_{2}`)
+measures the squared difference between the actual target variable, :math:`y`,
+and the predicted values. Minimizing this term alone could lead to overfitting,
+as the model may become too complex and sensitive to noise in the training data.
+
+To address overfitting, Ridge regularization adds a constraint, called a penalty
+term, (:math:`\\alpha \\| \\beta\\|^{2}_{2}`) to the loss function. This penalty
+term is the sum of the squares of the model's coefficients, multiplied by the
+regularization strength :math:`\\alpha`. By introducing this constraint, Ridge
+regularization discourages any single coefficient :math:`\\beta_{i}` from taking
+an excessively large value and encourages smaller and more evenly distributed
+coefficients. Higher values of :math:`\\alpha` force the coefficients towards
+zero. However, an excessively high :math:`\\alpha` can result in an underfit
+model that fails to capture important patterns in the data.
+
+Therefore, the regularized loss function combines the prediction accuracy term
+and the penalty term. By adjusting the regularization strength, practitioners
+can fine-tune the degree of constraint imposed on the weights, training a model
+capable of generalizing well to unseen data while avoiding overfitting.
+
+For a comparison between Ridge and Huber Regularization see
+:doc:`~examples.linear_model.plot_huber_vs_ridge`
 """
 
 # Author: Kornel Kielczewski -- <kornel.k@plusnet.pl>
 
-import matplotlib.pyplot as plt
-import numpy as np
+# %%
+# Purpose of this example
+# -----------------------
+# For the purpose of showing how Ridge regularization works, we will create two
+# data sets: non-noisy and noisy. Then we will train them on a range of
+# regularization strengths (:math:`\alpha`) and plot how the trained
+# coefficients and the mean squared error between those and the original values
+# behave as functions of the regularization strength.
+
+# %%
+# Ridge Regularization on non-noisy data set
+# ------------------------------------------
 
+# %%
+# Creating non-noisy data set
+#############################
+# We make a toy data set, that's suitable to detect regression. Please note that
+# in this example the data is non-noisy, hence it is possible to extract the
+# exact coefficients.
```

Comment:
Actually, this is not correct. When generating the data, we always have access to the true coefficients.
Here, not having noise means that we have `y = X @ w` while with noise we would have `y = X @ w + noise`.
So what we can mention here is that, given `X` and `y`, we expect our regression model to recover exactly the true coefficients `w`.

Reply:
Yes, thank you.